lib/src/clojure_lsp/feature/call_hierarchy.clj

```diff
@@ -42,19 +53,27 @@
         :parent-element cursor-element}
        db*)]))
 
+(defn ^:private parent-var-def
+  "Returns var def analysis element that surrounds the element at row/col (or
+  that is the element at that position)."
+  [root-zloc db filename row col]
+  (when-let [{:keys [row col]} (some-> root-zloc
+                                       (parser/to-pos row col)
+                                       (edit/find-var-definition-name-loc)
+                                       z/node
+                                       meta)]
+    (when-let [definition (q/find-element-under-cursor db filename row col)]
+      (when (or (identical? :var-definitions (:bucket definition))
+                (and (identical? :var-usages (:bucket definition))
+                     (:defmethod definition)))
+        definition))))
```

Comment:
This logic used to be in `edit`, even though it was only needed here. That was making `edit`, which is primarily about manipulating zlocs, too intimately familiar with the analysis. So, this moves the logic and implements it more efficiently.

Reply:
Great!